pageserver/src/tenant/blob_io.rs

```diff
@@ -83,41 +84,109 @@ impl<'a> BlockCursor<'a> {
     }
 }
 
+/// A wrapper of `VirtualFile` that allows users to write blobs.
 ///
-/// An implementation of BlobWriter to write blobs to anything that
-/// implements std::io::Write.
-///
-pub struct WriteBlobWriter<W> {
-    inner: W,
+/// If a `BlobWriter` is dropped, the internal buffer will be
+/// discarded. You need to call [`flush_buffer`](Self::flush_buffer)
+/// manually before dropping.
+pub struct BlobWriter<const BUFFERED: bool> {
+    inner: VirtualFile,
     offset: u64,
+    /// A buffer to save on read calls
+    buf: [u8; PAGE_SZ],
```

Comment:
This 8K is a bit wasteful if we're in BUFFERED=false mode (i.e. image layer writers).

Reply:
(see other comment about Cursor+Vec below -- either way, making this a Vec is probably sensible)

Reply:
I've adopted a `Vec` now.